Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionReferenceSearchProviders.cs

```diff
@@ -0,0 +1,98 @@
+                "Project-Wide Input Actions",
+                (obj) => "(Project-Wide Input Actions)",
+                () => InputActionImporter.LoadInputActionReferencesFromAsset(ProjectWideActionsAsset.GetOrCreate()));
+        }
+
+        private static SearchProvider CreateInputActionReferenceSearchProvider(string id, string displayName,
+            Func<Object, string> createItemFetchDescription, Func<IEnumerable<Object>> fetchAssets)
+        {
+            // Match icon used for sub-assets from importer for InputActionReferences.
+            // We assign description+label in FilteredSearch but also provide a fetchDescription+fetchLabel below.
+            // This is needed to support all zoom-modes for an unknown reason.
+            // Also, fetchLabel/fetchDescription and what is provided to CreateItem is playing different
+            // roles at different zoom levels.
+            var inputActionReferenceIcon = InputActionAssetIconLoader.LoadActionIcon();
+
+            return new SearchProvider(id, displayName)
+            {
+                priority = 25,
+                fetchDescription = FetchLabel,
+                fetchItems = (context, items, provider) => FilteredSearch(context, provider, FetchLabel, createItemFetchDescription,
+                    fetchAssets, "(Project-Wide Input Actions)"),
+                fetchLabel = FetchLabel,
+                fetchPreview = (item, context, size, options) => inputActionReferenceIcon,
+                fetchThumbnail = (item, context) => inputActionReferenceIcon,
+                toObject = (item, type) => item.data as Object,
+            };
+        }
+
+        // Custom search function with label matching filtering.
+        private static IEnumerable<SearchItem> FilteredSearch(SearchContext context, SearchProvider provider,
+            Func<Object, string> fetchObjectLabel, Func<Object, string> createItemFetchDescription, Func<IEnumerable<Object>> fetchAssets, string description)
+        {
+            foreach (var asset in fetchAssets())
+            {
+                var label = fetchObjectLabel(asset);
+                if (!label.Contains(context.searchText, System.StringComparison.InvariantCultureIgnoreCase))
+                    continue; // Ignore due to filtering
+                yield return provider.CreateItem(context, asset.GetInstanceID().ToString(), label, createItemFetchDescription(asset),
+                    null, asset);
+            }
+        }
+
+        // Note that this is overloaded to allow utilizing FetchLabel inside fetchItems to keep label formatting
+        // consistent between CreateItem and additional fetchLabel calls.
+        private static string FetchLabel(Object obj)
+        {
+            return obj.name;
+        }
+
+        private static string FetchLabel(SearchItem item, SearchContext context)
+        {
+            return FetchLabel((item.data as Object) !);
```

Comment:
This line doesn't compile.

Reply:
Could you clarify in what circumstances (editor version, OS, etc) ? I haven't seen any compile errors on my side and others, or CI.

Reply:
Sorry this was my mistake, I was trying to enable global actions on an earlier Unity Editor version that isn't supported.